### DISCUSSION TITLE
Ability to restrict category assignment to one category in any one group

### DISCUSSION THREAD

**Discussion by blue-dreamer** [ExpressionEngine/ExpressionEngine#2854] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Sometimes you only want authors to be able to select one category in a category group. For instance if I have a category group that describes colours where only one will ever be applicable.

**Describe the solution you'd like**
An option in Category group settings to specify if category selection can be limited to one item (SELECT element), or whether multiples are allowed (CHECKBOXES as now).

**Describe alternatives you've considered**
No alternatives exist for EE4/5 that I know of. There have been some 3rd party addons to do this but I believe it should be native.

**Teachability, Documentation, Adoption, Migration Strategy**
Uses existing docs, tags etc. Not sure how this would affect a current category group that has mutiples already assigned, presumably it would simply use the first available.

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by litzinger** [comment]
While you're at it you might as well add a max number of categories too, and make a minimum number of category selections required. E.g. they can choose up to 5, but a minimum of 3.

**Comment by blue-dreamer** [comment]
I've never had need to use max/min category selection but I agree it would be a useful option.

**Comment by nathanpitman** [comment]
This gets a vote from me too. In the past I have used [BW Required Category](https://devot-ee.com/add-ons/bw-required-category) to achieve this but it doesn't have EE3+ support.

**Comment by litzinger** [comment]
> I've never had need to use max/min category selection but I agree it would be a useful option.

I've had several cases where min/max category selections would be nice to have.

**Comment by intoeetive** [comment]
I'd like to get some ideas from people on how this could be implemented, if we decide to.

Like, would it be a per-channel setting? Or per-category-group setting? Or anything else?

**Comment by intoeetive** [comment]
We're getting this released in 7.4.0 via https://github.com/ExpressionEngine/ExpressionEngine/pull/3555
